test/component/profile_page_test.dart

```diff
@@ -49,96 +52,182 @@ void main() {
+      expect(centauriPoints, findsOneWidget);
+
+      //Check that the user account is displayed
+      final userAccount = find.byKey(UserAccount.userInfoKey);
+      expect(userAccount, findsOneWidget);
+
+      //Check that the tab is displayed
+      final tab = find.byKey(ProfilePage.tabKey);
+      expect(tab, findsOneWidget);
+    });
+
+    testWidgets("Tab working as expected", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      // Check that the post tab is displayed
+      final postTab = find.byKey(ProfilePage.postTabKey);
+      expect(postTab, findsOneWidget);
+
+      // Check that the comment tab is displayed
+      final commentTab = find.byKey(ProfilePage.commentTabKey);
+      expect(commentTab, findsOneWidget);
+
+      //Check that post column is displayed
+      final postColumn = find.byKey(ProfilePage.postColumnKey);
+      expect(postColumn, findsOneWidget);
+
+      // Tap on the comment tab
+      await tester.tap(commentTab);
+      await tester.pumpAndSettle();
+
+      // Check that the comment column is displayed
+      final commentColumn = find.byKey(ProfilePage.commentColumnKey);
+      expect(commentColumn, findsOneWidget);
+    });
+
+    testWidgets("Post popup working as expected", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      //Check that post column is displayed
+      final postColumn = find.byKey(ProfilePage.postColumnKey);
+      expect(postColumn, findsOneWidget);
+
+      //Check tab on the first post
+      final infoCardPost = find.byKey(InfoCardPost.infoCardPostKey);
+      expect(infoCardPost, findsWidgets);
+
+      // Tap on the first post
+      await tester.tap(infoCardPost.first);
+      await tester.pumpAndSettle();
+
+      // Check that the post popup is displayed
+      final postPopup = find.byType(PostPopUp);
+      expect(postPopup, findsOneWidget);
+
+      //Check that the title of the pop up is displayed
+      final postPopupTitle = find.byKey(PostPopUp.postPopUpTitleKey);
+      expect(postPopupTitle, findsOneWidget);
+
+      //Check that the description of the pop up is displayed
+      final postPopupDescription =
+          find.byKey(PostPopUp.postPopUpDescriptionKey);
+      expect(postPopupDescription, findsOneWidget);
+
+      //Check that the delete button is displayed
+      final postPopupDeleteButton =
+          find.byKey(PostPopUp.postPopUpDeleteButtonKey);
+      expect(postPopupDeleteButton, findsOneWidget);
+
+      //Check clicking on the delete button come back to the profile page
+      await tester.tap(postPopupDeleteButton);
+      await tester.pumpAndSettle();
+
+      //Check that the profile page is displayed
+      final profilePage = find.byType(ProfilePage);
+      expect(profilePage, findsOneWidget);
+    });
+
+    testWidgets("Tab working as expected", (tester) async {
```

Comment:
This test seems to regroup the two previous ones.
I think you could chose one and delete the other, to avoid code duplication.
If keeping the three tests is necessary, then you can create helper methods.

Reply:
Good Idea, I tried in 589b4e2 to regroup the tests together and create a common setup function `commonSetup`, while trying to keep test independencies.

Reply:
After discussing with @camillelnne, she suggested simply testing the common part of an already existing test (which already contains the testing of various UI elements) and making the other tests independent of this part.